dashboard/test/models/concerns/partial_registration_test.rb

```diff
@@ -4,93 +4,121 @@ class PartialRegistrationTest < ActiveSupport::TestCase
   # The PartialRegistration concern is only included in User, so we
   # are testing against User.
 
+  setup do
+    @session = {}
+  end
+
+  test 'persist_attributes does not push empty attributes into the cache' do
+    user = build :user
+    assert user.attributes.values.any?(&:nil?)
+
+    PartialRegistration.persist_attributes @session, user
+
+    cache_contents = CDO.shared_cache.read(PartialRegistration.cache_key(user))
+    deserialized = JSON.parse cache_contents
+    refute deserialized.values.any?(&:nil?)
+  end
+
+  test 'persist_attributes pushes less than 0.5KB into the cache for a sample user' do
+    user = build :user, :google_sso_provider
+    PartialRegistration.persist_attributes @session, user
+
+    cache_key = PartialRegistration.cache_key(user)
+    cache_contents = CDO.shared_cache.read(cache_key)
+    assert (cache_key + cache_contents).length < 512,
+      "Pushed #{cache_contents.length}B into the cache, expected less than 512B"
+  end
+
   test 'in_progress? is false when session has no user attributes' do
-    refute PartialRegistration.in_progress? fake_empty_session
+    refute PartialRegistration.in_progress? @session
+  end
+
+  test 'in_progress? is true when attributes have been persisted' do
+    PartialRegistration.persist_attributes @session, build(:user)
+    assert PartialRegistration.in_progress? @session
   end
 
-  test 'in_progress? is true when session has user attributes' do
-    assert PartialRegistration.in_progress? fake_session
+  test 'in_progress? becomes false if attributes are lost from the cache' do
+    user = build :user
+    PartialRegistration.persist_attributes @session, user
+    CDO.shared_cache.delete(PartialRegistration.cache_key(user))
+    refute PartialRegistration.in_progress? @session
   end
 
   test 'new_from_partial_registration raises unless a partial registration is available' do
     exception = assert_raise RuntimeError do
-      User.new_from_partial_registration fake_empty_session
+      User.new_from_partial_registration @session
     end
     assert_equal 'No partial registration was in progress', exception.message
   end
 
-  test 'new_from_partial_registration returns a User' do
-    user = User.new_from_partial_registration fake_session
-    assert_kind_of User, user
+  test 'new_from_partial_registration raises on a missing cache entry' do
+    user = build :student
+    PartialRegistration.persist_attributes @session, user
+    CDO.shared_cache.delete(PartialRegistration.cache_key(user))
+
+    exception = assert_raise RuntimeError do
+      User.new_from_partial_registration @session
+    end
+    assert_equal 'No partial registration was in progress', exception.message
```

Comment:
Is this the error we would present to the user in the rare scenario where memcached has evicted their cache entry? If so, should we tell them the error was temporary and that they should start over?